### PR TITLE
update verify-shellcheck to v0.7.1, fix nits, multi-arch digest pinning, fix new lint errors

### PR DESCRIPTION
**Pull request by BenTheElder**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

- fixes the pass / fail messages in `hack/verify-shellcheck.sh`
- updates to current shellcheck stable (v0.7.0 -> v0.7.1)
- switches to a multi-arch digest instead of an amd64 digest (thanks @jonjohnsonjr)
-  fixes new lint errors (trivial unquoted variable expansion issues)
- enable `DOCKER` env override for docker CLI, like other kubernetes scripts

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
/sig testing

### CODE REVIEW

File: hack/verify-shellcheck.sh [hunk 1]
```diff
@@ -27,9 +27,9 @@ source "${KUBE_ROOT}/hack/lib/util.sh"
 
 # required version for this script, if not installed on the host we will
 # use the official docker image instead. keep this in sync with SHELLCHECK_IMAGE
-SHELLCHECK_VERSION="0.7.0"
+SHELLCHECK_VERSION="0.7.1"
 # upstream shellcheck latest stable image as of October 23rd, 2019
-SHELLCHECK_IMAGE="koalaman/shellcheck-alpine:v0.7.0@sha256:24bbf52aae6eaa27accc9f61de32d30a1498555e6ef452966d0702ff06f38ecb"
+SHELLCHECK_IMAGE="docker.io/koalaman/shellcheck-alpine:v0.7.1@sha256:d6147f30864ddb7c9cf983fc277d345bc315e798e309ddf70062b194843ee252"
```
Comment (BenTheElder) [inline hack/verify-shellcheck.sh:32]: :shrug

**Comment by BenTheElder** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/99905/pull-kubernetes-node-e2e/1368312774706335744
https://github.com/kubernetes/kubernetes/issues/98649 again

File: hack/verify-shellcheck.sh [hunk 2]
```diff
@@ -105,23 +108,23 @@ if ${HAVE_SHELLCHECK}; then
...
+  echo 'Congratulations! All shell files are passing lint :-)'
 else
   {
     echo
     echo 'Please review the above warnings. You can test via "./hack/verify-shellcheck.sh"'
     echo 'If the above warnings do not make sense, you can exempt this warning with a comment'
     echo ' (if your reviewer is okay with it).'
     echo 'In general please prefer to fix the error, we have already disabled specific lints'
-    echo ' that the project chooses to ignire.'
+    echo ' that the project chooses to ignore.'
```
Comment (liggitt) [inline hack/verify-shellcheck.sh:127]: shellcheck, meet spellcheck :)

File: test/cmd/core.sh [hunk 3]
```diff
@@ -560,7 +560,7 @@ run_pod_tests() {
   resourceVersion=$(kubectl get "${kube_flags[@]}" pod valid-pod -o go-template='{{ .metadata.resourceVersion }}')
   ((resourceVersion+=100))
   # Command
-  kubectl patch "${kube_flags[@]}" pod valid-pod -p='{"spec":{"containers":[{"name": "kubernetes-serve-hostname", "image": "nginx"}]},"metadata":{"resourceVersion":"'$resourceVersion'"}}' 2> "${ERROR_FILE}" || true
+  kubectl patch "${kube_flags[@]}" pod valid-pod -p='{"spec":{"containers":[{"name": "kubernetes-serve-hostname", "image": "nginx"}]},"metadata":{"resourceVersion":"'"$resourceVersion"'"}}' 2> "${ERROR_FILE}" || true
```
Comment (liggitt) [inline test/cmd/core.sh:563]: took me a minute to parse this, but it evaluates to:
```
-p=
'{"spec":{"containers":[{"name": "kubernetes-serve-hostname", "image": "nginx"}]},"metadata":{"resourceVersion":"'
"$resourceVersion"
'"}}'
```

Comment (BenTheElder) [reply test/cmd/core.sh:563]: We could instead write the whole thing as a double quote string with escape sequences, but this change is smaller and "normal" shell (adjacent strings are concatenated even if alternating quote types, it's pretty common for concatenating literals with variable expansion like this)

File: hack/verify-shellcheck.sh (hunk 2 above)
Comment (BenTheElder) [reply hack/verify-shellcheck.sh:127]: we could do with a more complete spell checker.

Comment (liggitt) [reply hack/verify-shellcheck.sh:127]: Please, no

**Pull request merged**